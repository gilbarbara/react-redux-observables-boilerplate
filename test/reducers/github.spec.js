import reducers from 'reducers';
import { ActionTypes } from 'constants/index';

describe('Github', () => {
  it('should return the initial state', () => {
    expect(reducers.github(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.FETCH_POPULAR_REPOS_REQUEST}`, () => {
    expect(reducers.github(undefined, { type: ActionTypes.FETCH_POPULAR_REPOS_REQUEST })).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.FETCH_POPULAR_REPOS_SUCCESS}`, () => {
    expect(reducers.github(undefined, {
      type: ActionTypes.FETCH_POPULAR_REPOS_SUCCESS,
      payload: { data: [{ id: 1 }] },
    }))
      .toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.FETCH_POPULAR_REPOS_FAILURE}`, () => {
    expect(reducers.github(undefined, {
      type: ActionTypes.FETCH_POPULAR_REPOS_FAILURE,
      payload: { message: 'ajax error 404', status: 404 },
    })).toMatchSnapshot();
  });
});
