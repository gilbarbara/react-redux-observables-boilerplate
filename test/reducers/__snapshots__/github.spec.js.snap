// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github should handle FETCH_POPULAR_REPOS_FAILURE 1`] = `
Object {
  "popularRepos": Object {
    "data": Array [],
    "message": "ajax error 404",
    "status": "error",
  },
}
`;

exports[`Github should handle FETCH_POPULAR_REPOS_REQUEST 1`] = `
Object {
  "popularRepos": Object {
    "data": Array [],
    "message": "",
    "status": "running",
  },
}
`;

exports[`Github should handle FETCH_POPULAR_REPOS_SUCCESS 1`] = `
Object {
  "popularRepos": Object {
    "data": Array [
      Object {
        "id": 1,
      },
    ],
    "message": "",
    "status": "loaded",
  },
}
`;

exports[`Github should return the initial state 1`] = `
Object {
  "popularRepos": Object {
    "data": Array [],
    "message": "",
    "status": "idle",
  },
}
`;
